Shared/ViewModels/UserSignInViewModel.swift

```diff
@@ -73,59 +63,176 @@ final class UserSignInViewModel: ViewModel {
+              let id = response.user?.id else { throw JellyfinAPIError("Missing user data from network call") }
 
-        await MainActor.run {
-            quickConnectEnabled = isEnabled ?? false
+        if let existingUser = try? SwiftfinStore.dataStack.fetchOne(
+            From<UserModel>(),
+            [Where<UserModel>(
+                "id == %@",
+                id
+            )]
+        ) {
+            throw SwiftfinStore.Error.existingUser(existingUser.state)
+        }
+
+        guard let storedServer = try? SwiftfinStore.dataStack.fetchOne(
+            From<SwiftfinStore.Models.StoredServer>(),
+            [
+                Where<SwiftfinStore.Models.StoredServer>(
+                    "id == %@",
+                    server.id
+                ),
+            ]
+        )
+        else { fatalError("No stored server associated with given state server?") }
+
+        let user = try SwiftfinStore.dataStack.perform { transaction in
+            let newUser = transaction.create(Into<UserModel>())
+
+            newUser.accessToken = accessToken
+            newUser.appleTVID = ""
+            newUser.id = id
+            newUser.username = username
+
+            let editServer = transaction.edit(storedServer)!
+            editServer.users.insert(newUser)
+
+            return newUser.state
         }
+
+        return user
     }
 
-    func startQuickConnect() -> AsyncStream<QuickConnectResult> {
-        Task {
+    // MARK: - Quick Connect
 
-            let initiatePath = Paths.initiate
-            let response = try? await client.send(initiatePath)
+    /// The typical quick connect lifecycle is as follows:
+    /// 1. User clicks quick connect
+    /// 2. We fetch a secret and code from the server
+    /// 3. Display the code to user, poll for authentication from server using secret
+    /// 4. User enters code to the server
+    /// 5. Authentication poll succeeds with another secret, use secret to log in
+
+    @Published
+    private(set) var quickConnectEnabled = false
+    /// We want to tell the monitor task when we're ready to poll for authentication. Without this, we may encounter
+    /// a race condition where the monitor task expects the secret before it's fetched and exits prematurely.
+    @Published
+    private(set) var quickConnectStatus: QuickConnectStatus?
+
+    private let quickConnectPollTimeoutSeconds: UInt64 = 5
+
+    /// We don't use Timer.scheduledTimer because checking quick connect status is async. We only want to start the next
+    /// poll when the current has finished.
+    private var quickConnectPollTask: Task<Void, Never>?
+    /// When two quick connect tasks are started for whatever reason, cancelling the repeating task above seems to fail
+    /// and the attempted cancelled task seems to continue to spawn more repeating tasks. We ensure only a single
+    /// poll task continues to live with this ID.
+    private var quickConnectPollTaskID: UUID?
```

Comment:
Actually having thought about this a bit more, remove this and add another `QuickConnectStatus.authorized` and don't start quick connect again if authorized. We aren't running into the issue that we are starting quick connect during auth checks and if we are, something is instead wrong with the views. If we ever run into the problem where we want to start quick connect again while authorized, we can deal with that later?

That's just a bit cleaner and we don't have to worry about tracking an ID, and instead we can cancel the `quickConnectPollTask` when stopping. In the task we can check if the task was canceled before we poll again.